services/src/main/java/org/keycloak/forms/login/freemarker/FreeMarkerLoginFormsProvider.java

```diff
@@ -209,6 +209,9 @@ protected Response createResponse(LoginFormsPages page) {
                 BrokeredIdentityContext brokerContext = (BrokeredIdentityContext) this.attributes.get(IDENTITY_PROVIDER_BROKER_CONTEXT);
                 String idpAlias = brokerContext.getIdpConfig().getAlias();
                 idpAlias = ObjectUtil.capitalize(idpAlias);
+                if (!ObjectUtil.isBlank(brokerContext.getIdpConfig().getDisplayName())) {
+                    idpAlias = brokerContext.getIdpConfig().getDisplayName();
+                }
```

Comment:
This would give misleading information to the `attributes`. `idpAlias` must contain identity provider alias. You can however add another attribute, e.g. idpDisplayName, that would be display name (if set) or alias (if displayname unset), and use that in the template.

Reply:
This part is correct, yet see below